src/libs/actions/Welcome.ts

```diff
@@ -147,19 +159,17 @@ function show({routes, showCreateMenu = () => {}, showPopoverMenu = () => false}
                 Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(workspaceChatReport.reportID));
             }
 
-            // If showPopoverMenu exists and returns true then it opened the Popover Menu successfully, and we can update isFirstTimeNewExpensifyUser
-            // so the Welcome logic doesn't run again
-            if (showPopoverMenu?.()) {
-                isFirstTimeNewExpensifyUser = false;
-            }
+            // New user has been redirected to their workspace chat, and we won't show them the engagement modal.
+            // So we update isFirstTimeNewExpensifyUser to prevent the Welcome logic from running again
+            isFirstTimeNewExpensifyUser = false;
 
             return;
         }
 
         // If user is not already an admin of a free policy and we are not navigating them to their workspace or creating a new workspace via workspace/new then
-        // we will show the create menu.
-        if (!Policy.isAdminOfFreePolicy(allPolicies ?? undefined) && !isDisplayingWorkspaceRoute) {
-            showCreateMenu();
+        // we will show the engagement modal.
+        if (!Policy.isAdminOfFreePolicy(allPolicies ?? undefined) && !isExitingToWorkspaceRoute && !hasSelectedChoice && !hasDismissedModal && Object.keys(allPolicies ?? {}).length === 1) {
```

Comment:
@stitesExpensify Why do we need to check ```Object.keys(allPolicies ?? {}).length === 1``` in here?

Reply:
Pretty sure it's because everyone should at least have a personal policy. So if you have some other policy on top of that then we're not trying to engage you as a new user.

Reply:
^ This